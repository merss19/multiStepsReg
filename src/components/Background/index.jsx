"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var data_1 = require("./data");
var Background = function () {
    return (<div className='entry__bg'>
				{data_1.bg}
			</div>);
};
exports.default = Background;
