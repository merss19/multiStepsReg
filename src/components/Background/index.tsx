import * as React from 'react';
import {bg} from './data';

const Background: React.SFC<{}> = () => {
		return (
			<div className='entry__bg'>
				{bg}
			</div>
		)
};

export default Background;