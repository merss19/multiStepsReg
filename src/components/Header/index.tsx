import Header from './main'
export {Header}