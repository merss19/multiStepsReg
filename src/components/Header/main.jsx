"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var Button_1 = require("components/Button");
var react_router_dom_1 = require("react-router-dom");
var storage_1 = require("tools/storage");
var Header = /** @class */ (function (_super) {
    __extends(Header, _super);
    function Header() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    Header.prototype.logOut = function () {
        storage_1.storage.remove('token', { path: '/' });
        storage_1.storage.remove('program', { path: '/' });
        storage_1.storage.remove('promoName', { path: '/' });
        storage_1.storage.removeState('auth');
    };
    Header.prototype.render = function () {
        return (<div className='header header--hero'>
        <div className='grid header__inner'>
          <div className='grid__cell header__burger'>
            <span className='header__ico-burger'>
              <svg className='svg-icon ico-burger'>
                <use xlinkHref="#ico-burger"></use>
              </svg>
            </span>
          </div>

          <h1 className='grid__cell header__logo'>
            Ясегодня
            <svg className='svg-icon ys-logo-web'>
              <use xlinkHref="#ys-logo-web"></use>
            </svg>
          </h1>

          <div className='grid__cell header__right-side'>
            <Button_1.Button type='button' styleBtn={Button_1.ButtonTypes.info}>
              <react_router_dom_1.Link to='/' onClick={this.logOut.bind(this)}>Выход</react_router_dom_1.Link>
            </Button_1.Button>
          </div>
        </div>
      </div>);
    };
    return Header;
}(React.Component));
exports.Header = Header;
exports.default = Header;
