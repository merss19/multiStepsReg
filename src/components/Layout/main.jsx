"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var Background_1 = require("../Background");
var LogoLink_1 = require("../LogoLink");
var SVGInjector = require("svg-injector");
var Header_1 = require("components/Header");
var Layout = /** @class */ (function (_super) {
    __extends(Layout, _super);
    function Layout() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    Layout.prototype.componentDidMount = function () {
        this.svgInject();
    };
    Layout.prototype.svgInject = function () {
        var sprite = {
            body: document.querySelector('body'),
            img: document.createElement('img')
        };
        sprite.img.className = 'injected-svg';
        sprite.img.id = 'svg-inject-me';
        sprite.img.src = process.env.PUBLIC_URL + '/img/symbol-sprite.svg';
        sprite.body.appendChild(sprite.img);
        var mySVGsToInject = document.querySelector('#svg-inject-me');
        SVGInjector(mySVGsToInject);
    };
    Layout.prototype.renderLogo = function () {
        if (!this.props.notLogo) {
            return (<div className='grid entry-header'>
          <div className='grid__cell todayme-logo'>
            <LogoLink_1.default />
          </div>
        </div>);
        }
    };
    Layout.prototype.render = function () {
        var _a = this.props, children = _a.children, type = _a.type;
        if (type === 'main') {
            return (<Header_1.Header />);
        }
        else {
            return (<div className='layout layout--entry'>
          {this.renderLogo()}
          {children}
          <Background_1.default />
        </div>);
        }
    };
    Layout.defaultProps = {
        notLogo: false,
        type: 'signUP'
    };
    return Layout;
}(React.Component));
exports.Layout = Layout;
exports.default = Layout;
