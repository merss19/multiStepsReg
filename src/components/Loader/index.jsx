"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var Button_1 = require("components/Button");
var react_router_dom_1 = require("react-router-dom");
exports.Loader = function (_a) {
    var text = _a.text, isError = _a.isError;
    return (<div className='loading text-center'>
      {isError ?
        <div className='loading__desc text-center pt20'>
          <h2 className='pb20'>{text}</h2>
          <div className='pt10 pb10'>
            <p>Что-то пошло не так.</p>
            <p>Напишите нам в чат тех поддержки!</p>
          </div>

          <Button_1.Button type='button' styleBtn={Button_1.ButtonTypes.info}>
            <react_router_dom_1.Link to="/">Продолжить</react_router_dom_1.Link>
          </Button_1.Button>
        </div>
        :
            <div className='loader loader--main'/>}
    </div>);
};
exports.Loader.defaultProps = {
    text: 'Загружается',
};
