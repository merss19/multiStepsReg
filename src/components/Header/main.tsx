import * as React from 'react';
import {Button, ButtonTypes} from "components/Button";
import {Link} from "react-router-dom";
import {storage} from "tools/storage";

interface OwnProps {
}

export class Header extends React.Component<OwnProps, {}> {

  logOut() {
    storage.remove('token', {path: '/'});
    storage.remove('program', {path: '/'});
    storage.remove('promoName', {path: '/'});
    storage.removeState('auth');
  }

  render() {
    return (
      <div className='header header--hero'>
        <div className='grid header__inner'>
          <div className='grid__cell header__burger'>
            <span className='header__ico-burger'>
              <svg className='svg-icon ico-burger'>
                <use xlinkHref="#ico-burger"></use>
              </svg>
            </span>
          </div>

          <h1 className='grid__cell header__logo'>
            Ясегодня
            <svg className='svg-icon ys-logo-web'>
              <use xlinkHref="#ys-logo-web"></use>
            </svg>
          </h1>

          <div className='grid__cell header__right-side'>
            <Button
              type='button'
              styleBtn={ButtonTypes.info}
            >
              <Link to='/' onClick={this.logOut.bind(this)}>Выход</Link>
            </Button>
          </div>
        </div>
      </div>
    );
  }
}

export default Header;