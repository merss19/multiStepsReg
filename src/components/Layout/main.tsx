import * as React from 'react';
import Background from '../Background';
import LogoLink from '../LogoLink'
import * as SVGInjector from 'svg-injector'
import {Header} from 'components/Header';

interface Sprite {
  readonly body: any
  readonly img: HTMLImageElement
}
interface OwnProps {
  notLogo?: boolean
  type: 'main' | 'signUP'
}
export class Layout extends React.Component<OwnProps, {}> {

  static defaultProps = {
    notLogo: false,
    type:'signUP'
  };

  componentDidMount() {
    this.svgInject()
  }

  svgInject(): void {
    const sprite: Sprite = {
      body: document.querySelector('body'),
      img: document.createElement('img')
    };
    sprite.img.className = 'injected-svg';
    sprite.img.id = 'svg-inject-me';
    sprite.img.src = process.env.PUBLIC_URL + '/img/symbol-sprite.svg';
    sprite.body.appendChild(sprite.img);
    let mySVGsToInject: any = document.querySelector('#svg-inject-me');
    SVGInjector(mySVGsToInject);
  }

  renderLogo() {
    if (!this.props.notLogo) {
      return (
        <div className='grid entry-header'>
          <div className='grid__cell todayme-logo'>
            <LogoLink/>
          </div>
        </div>
      )
    }
  }

  render() {
    const {children, type} = this.props;
    if(type === 'main'){
      return (<Header/>);
    } else {
      return (
        <div className='layout layout--entry'>
          {this.renderLogo()}
          {children}
          <Background/>
        </div>
      );
    }
  }
}

export default Layout;