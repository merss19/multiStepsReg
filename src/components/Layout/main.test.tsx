import {shallow, mount} from 'enzyme'
import Layout from './main'
import * as React from 'react'
import toJson from 'enzyme-to-json'
import Background from '../Background'
import LogoLink from '../LogoLink'
import * as sinon from 'sinon'
import {configure} from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import {Loader} from '../Loader/index';
import {Header} from '../Header/main';

configure({adapter: new Adapter()});
/*process.env.NODE_ENV = 'test'*/
let notLogo;
let type;
const start = () => {
  beforeEach(() => {
    notLogo = false;
    type = 'signUp';
  });
  const props = {
    notLogo: notLogo,
    type: type
  };
  const buildComponent = () => {
    return shallow(<Layout {...props}><Loader /></Layout>)
  };
  return {
    props,
    buildComponent
  }
};
describe('Component Layout', () => {
  it('renders component', () => {
    const component = start().buildComponent();
    expect(component).toHaveLength(1);
  });
  it('renders correctly', () => {
    const tree = toJson(start().buildComponent());
    expect(tree).toMatchSnapshot();
  });
  it('check prop notLogo', () => {
    const component = start().buildComponent();
    expect(component.find('.entry-header').exists()).toBe(true);
    expect(component.find(LogoLink).exists()).toBe(true);
    notLogo = true;
    component.setProps({notLogo: notLogo});
    expect(component.find('.entry-header').exists()).toBe(false);
    expect(component.find(LogoLink).exists()).toBe(false);
    notLogo = undefined;
    component.setProps({notLogo: notLogo});
    expect(component.find('.entry-header').exists()).toBe(true);
    expect(component.find(LogoLink).exists()).toBe(true);
  });
  it('renders the Background', () => {
    const component = start().buildComponent();
    expect(component.find(Background).exists()).toBe(true);
  });
  it('render children component', () => {
    const component = start().buildComponent();
    expect(component.find(Loader).exists()).toBe(true);
  });
  it('render Header', () => {
    type = 'main';
    const component = start().buildComponent();
    expect(component.find(Header).exists()).toBe(true);
  });
  it('call svgInject', () => {
    const spy = sinon.spy(Layout.prototype, 'svgInject');
    const component = start().buildComponent();
    expect(spy.calledOnce).toEqual(true);
  });
});
/*
describe('Component Layout DOM', () => {
	const component = mount(<Layout>
		<div>test</div>
	</Layout>)
	it('children component defined', () => {
		expect(component.props().children).toBeDefined()
	})
	it('do not render logo', () => {
		const component = mount(<Layout notLogo={true} />)
		expect(component.find(LogoLink).exists()).toBe(false)
	})
	it('calls componentDidMount', () => {
		const spy = sinon.spy(Layout.prototype, 'componentDidMount')
		const wrapper = mount(<Layout />)
		expect(spy.calledOnce).toEqual(true)
	})
/!*	it('calls svjInject', () => {
		const spy = sinon.spy(Layout.prototype, 'svgInject')
		const wrapper = mount(<Layout />)
		expect(spy.calledOnce).toEqual(true)
	})*!/
})*/
