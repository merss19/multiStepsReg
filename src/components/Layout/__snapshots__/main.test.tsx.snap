// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Layout renders correctly 1`] = `
<div
  className="layout layout--entry"
>
  <div
    className="grid entry-header"
  >
    <div
      className="grid__cell todayme-logo"
    >
      <LogoLink />
    </div>
  </div>
  <Component
    text="Загружается"
  />
  <Background />
</div>
`;
